data/data/openstack/main.tf

```diff
@@ -72,3 +72,20 @@ module "topology" {
   trunk_support       = var.openstack_trunk_support
   octavia_support     = var.openstack_octavia_support
 }
+
+resource "openstack_images_image_v2" "base_image" {
+  count = var.openstack_base_image_url == "" ? 0 : 1
+
+  name             = var.openstack_base_image
```

Comment:
So we can still upload a second image with the same name?
Is there perhaps a way to force terraform to overwrite an image with the same name?

Reply:
When installer generates a name it is always unique (infraID + "-rhcos"), so, no we can't create a second image with the same name here.
Also users can set a custom glance image with OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE variable. In the case of multiple images with the same name deployments should fail early because I removed "most_recent" flag for the terraform data.

Reply:
So what happens if the user overrides the base image with `OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE` ? Will it upload a duplicate image?

We know this is going to break during the deployment. We should prevent uploading image in glance if there is already an image with the same name.

Reply:
If the user overrides OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE then var.openstack_base_image_url will be empty, and count is 0. In other words we won't create an image if the variable is set.

Reply:
This should be covered by a unit test then because we have no guarantee that `openstack_base_image_url` will remain empty when `OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE` is set.

IMO it's better not to rely on the installer's internal, and implement a quick check that prevents from uploading a new image under the same name.